src/main/java/core/basesyntax/converter/DataConverterImpl.java

```
@@ -0,0 +1,66 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String COMMA_SEPARATOR = ",";
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> report) {
+        Objects.requireNonNull(report, "report must not be null");
+
+        return IntStream.range(0, report.size())
+                .mapToObj(i -> {
+                    String line = report.get(i);
+                    int lineNumber = i + 1;
+                    if (line == null || line.isBlank()) {
+                        throw new IllegalArgumentException("Line "
+                                + lineNumber + " is empty or null");
+                    }
+                    String[] parts = line.split(COMMA_SEPARATOR, -1);
+                    if (parts.length != 3) {
+                        throw new IllegalArgumentException(
+                                "Line " + lineNumber 
+                                        + " must contain exactly 3 fields but has "
+                                        + parts.length + ": \"" + line + "\"");
+                    }
+                    String opToken = parts[0].trim();
+                    String fruit = parts[1].trim();
```

You trim the fruit token but do not validate it. The CSV handling requirements state: "Validate that fruit name is non-null, non-blank (after trimming)." After `String fruit = parts[1].trim();` add a check such as `if (fruit.isBlank()) { throw new IllegalArgumentException("Empty fruit name on line " + lineNumber + ": \"" + line + "\""); }` so invalid input fails fast and with a line-aware message (include lineNumber and line content).